File: frontend/src/types/shopItem.ts
export enum ShopItemVisualType {
  AvatarFrame = 0,
  ChatFrame = 1,
  NameBadge = 2,
}

export enum ShopItemTier {
  Basic = 0,
  Advanced = 1,
  Elite = 2,
  Legendary = 3,
}

export interface ShopItem {
  id: number
  name: string
  description?: string
  costInPoints: number
  iconUrl: string
  usageDurationDays: number
  visualType: ShopItemVisualType
  tier: ShopItemTier
  isDefault: boolean
  configJson?: string
  createdAt?: string
  updatedAt?: string | null
}

export interface CreateShopItemRequest {
  name: string
  description?: string
  costInPoints: number
  iconUrl?: string
  usageDurationDays: number
  visualType: ShopItemVisualType
  tier: ShopItemTier
  isDefault?: boolean
  configJson?: string
}

export interface UpdateShopItemRequest extends CreateShopItemRequest {
  id: number
}

export interface GetPaginatedShopItemsRequest {
  pageNumber?: number
  pageSize?: number
  keyword?: string
}

export interface PaginatedShopItemsResponse {
  items: ShopItem[]
  totalItems: number
  pageIndex: number
  pageSize: number
  totalPages: number
  hasPreviousPage: boolean
  hasNextPage: boolean
}

export interface UploadShopItemIconResponse {
  iconUrl: string
}

File: backend/Models/Entities/ShopItem.cs
using OnlineClassroomManagement.Helper.Constants;

namespace OnlineClassroomManagement.Models.Entities
{
    public class ShopItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int CostInPoints { get; set; }
        public string IconUrl { get; set; }
        public string ConfigJson { get; set; } = "{}";
        public int UsageDurationDays { get; set; } = 30;
        public ShopItemVisualType VisualType { get; set; } = ShopItemVisualType.NameBadge;
        public ShopItemTier Tier { get; set; } = ShopItemTier.Basic;
        public bool IsDefault { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}
